### DISCUSSION TITLE
How are you supposed to deal with wrapper scripts?

### DISCUSSION THREAD

**Discussion by NetSysFire** [netblue30/firejail#5096] [Q&A]
There are a couple of applications I have installed which are using wrapper scripts. So `/usr/bin/program` is the wrapper bash script which will eventually execute `/usr/bin/program.bin`, which is the actual binary. Usually with positional arguments the wrapper script found out somehow.

I would like to really avoid sandboxing both the wrapper script and actual binary because that would mean it has access to a lot of shell stuff for more code execution, so the wrapper script can do what it wants but the binary should be run sandboxed.

The two specific programs are:

1. The eternal lands client (https://github.com/raduprv/Eternal-Lands/blob/402097b4213fb0f036a392032695b500d30dc79d/pkgfiles/eternallands#L286)
2. The ember client for worldforge (https://github.com/worldforge/ember/blob/master/tools/ember.tmpl#L45)

The problem is that they appear to use absolute paths and that would circumvent the symlinks created by `firecfg`. I assume it would work fine if they do not use absolute paths and the user has used firecfg to create the symlinks.

Would the proper solution be to actually fix that upstream?

**Comment by rusty-snake** [comment]
If you do not want to run `firejail -> program -> program.bin` and instead `program -> firejail -> program.bin, you likely need to copy&patch the wrapper scripts.
